lib/Twig/Node.php

```diff
@@ -213,6 +213,10 @@ public function getNode($name)
      */
     public function setNode($name, $node = null)
     {
+        if (!$node instanceof Twig_NodeInterface) {
```

Comment:
the check should be on ``Twig_Node`` instead, as the interface is deprecated and people must extend the base class